device-code-flow/app.ps1

```diff
@@ -0,0 +1,66 @@
+# A PowerShell application that demonstrates how to use the
+# Device Code flow to make an API call to Microsoft Graph.
+
+# 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+$ClientId = ""
+
+# 'Directory (tenant) ID' of app registration in Azure portal - this value is a GUID
+$TenantId = ""
+
+# Configure $DeviceCodeRequestParams for the desired $TenantId and $ClientId
+$DeviceCodeRequestParams = @{
+    Method = "POST"
+    Uri    = "https://login.microsoftonline.com/$TenantId/oauth2/devicecode"
+    Body   = @{
+        resource  = "https://graph.microsoft.com"
+        client_id = $ClientId
+    }
+}
```

Comment:
Typically, we'd expect users to use MSAL.NET from PowerShell.  One of PowerShell's key features is to use .NET assemblies as part of its execution, so that you don't need to replicate logic within PowerShell that already has a .NET SDK.  Even our unsupported MSAL.PS module uses MSAL.NET under the covers.  There is just too much error handling, token caching, etc that is being left on the table when trying to talk to Azure AD without the msal.net library.

Any reason to avoid it that you came across?  Even the Azure SDK for powershell uses `Azure.Identity` which itself uses MSAL.NET.

Reply:
Resolving this comment since we have a direction here.